src/services/panel.ts

```diff
@@ -282,7 +282,11 @@ export function getResourceQueryRunner(queries: LokiQuery[], queryRunnerOptions?
   });
 }
 
-export function getQueryRunner(queries: LokiQuery[], queryRunnerOptions?: Partial<QueryRunnerState>) {
+export function getQueryRunner(
```

Comment:
maybe worth splitting `getQueryRunner` into getLogsPanelQueryRunner and getPanelQueryRunner?

Reply:
Actually while this might introduce some refactor bloat to this PR, this method returning different max lines functionality depending on the existence of an optional sceneRef is a bit of a foot gun for future devs.

Although reverting these changes to getQueryRunner and creating a new method which is (only?) used in the serviceScene isn't that big of a change and should be much clearer for future devs